### ISSUE TITLE
IO Fencing enforcement

### ISSUE THREAD

**Issue by Jgoldick** [container-storage-interface/spec#46]
Shouldn't an explicit IO Fencing requirement be added as a plugin's responsibility when MULTI_NODE_MULTI_WRITER is not set? Historically one cannot rely upon the original node to stop all writes even after the CO has decided to move a container+Volume to a new node. A lack of explicit IO fencing has historically the cause of much data corruption. A reasonable question is whether this responsibility belongs to the ControllerPublishVolume or NodePublishVolume. I don't believe we can trust NodeUnpublishVolume to guarantee that no pending writes will ever leave the kernel of the node as this behavior differs even among Linux variants. Storage systems generally have the ability to enforce this requirement because of the inability to trust hosts.

**Comment by jdef** [comment]
Yes, this sounds like it should be the responsibility of a plugin.
